src/connection.rs

```diff
@@ -74,6 +118,54 @@ impl Write for Connection {
     fn flush(&mut self) -> io::Result<()> {
         match self.stream {
             Stream::TcpStream(ref mut stream) => stream.flush(),
+            Stream::UdpSocket(ref mut stream) => { 
+                //udp header is 8 bytes in the begining of each datagram
+                let mut udp_header: Vec<u8> = Vec::new();
+                udp_header.write_u16::<BigEndian>(stream.request_id)?; // request id to uniquely identify response for this request
+                udp_header.write_u16::<BigEndian>(0)?; // 0 indicates this is the first datagram for this request
+                udp_header.write_u16::<BigEndian>(1)?; // total datagrams in this request (requests can only be 1 datagram long)
+                udp_header.write_u16::<BigEndian>(256)?; // reserved bytes (tested to also work with zero)
+                udp_header.reverse();
+                udp_header.iter().for_each(|i| stream.write_buf.insert(0, *i));
```

Comment:
Insert a element one by one to a vector will cost more CPU cycles, and vector's length will alloc multiple times (just guessing, not saw rust's vector's implementation, but is's common in any languages). We should use some stuff like `extend`, may be `extend` or `extend_from_slice`.

Reply:
I was looking for a prepend method but rust doesn't have any. So will use [splice](https://stackoverflow.com/a/47037876/1402140).